### PR TITLE
config files: update their header to reflect that they can be install…

### PR DESCRIPTION
**Pull request by fbuihuu**
…ed in /usr

Follow-up for c76f2fb0e59340222ce21f85c17d384c114db9de.

### CODE REVIEW

**Comment by fbuihuu** [comment]
@YHNdnzj let me know if you really want that all directory paths end with '/', I'll update the patch.

**Comment by YHNdnzj** [comment]
> @YHNdnzj let me know if you really want that all directory paths end with '/', I'll update the patch.

Yeah, I want /etc/ and /usr/ to be suffixed too.

**Comment by fbuihuu** [comment]
> Yeah, I want /etc/ and /usr/ to be suffixed too.

ok

**Comment by fbuihuu** [comment]
@YHNdnzj done.

I also reverted the changes in udev.conf since udevd currently supports reading its configuration only from /etc/udev/udev.conf.

**Review by bluca — APPROVED** [review]

File: src/journal-remote/journal-remote.conf.in [hunk 1]
```diff
@@ -6,9 +6,11 @@
 #  any later version.
 #
 # Entries in this file show the compile time defaults. Local configuration
-# should be created by either modifying this file, or by creating "drop-ins" in
-# the journal-remote.conf.d/ subdirectory. The latter is generally recommended.
-# Defaults can be restored by simply deleting this file and all drop-ins.
+# should be created by either modifying this file (or a copy of it placed in
+# /etc/ if the original file is shipped in /usr/), or by creating "drop-ins" in
+# /etc/systemd/journal-remote.conf.d/ directory. The latter is generally
+# recommended.  Defaults can be restored by simply deleting the main
```
Comment (YHNdnzj) [inline src/journal-remote/journal-remote.conf.in]: Duplicate space?

File: src/journal-remote/journal-upload.conf.in [hunk 2]
```diff
@@ -6,9 +6,11 @@
 #  any later version.
 #
 # Entries in this file show the compile time defaults. Local configuration
-# should be created by either modifying this file, or by creating "drop-ins" in
-# the journal-upload.conf.d/ subdirectory. The latter is generally recommended.
-# Defaults can be restored by simply deleting this file and all drop-ins.
+# should be created by either modifying this file (or a copy of it placed in
+# /etc/ if the original file is shipped in /usr/), or by creating "drop-ins" in
+# /etc/systemd/journal-upload..conf.d/ directory. The latter is generally
```
Comment (YHNdnzj) [inline src/journal-remote/journal-upload.conf.in]:
```suggestion
# /etc/systemd/journal-upload.conf.d/ directory. The latter is generally
```

File: src/journal-remote/journal-upload.conf.in [hunk 3]
```diff
@@ -6,9 +6,11 @@
...
+# recommended.  Defaults can be restored by simply deleting the main
```
Comment (YHNdnzj) [inline src/journal-remote/journal-upload.conf.in]: Duplicate space also here

File: src/timesync/timesyncd.conf.in [hunk 4]
```diff
@@ -6,9 +6,11 @@
 #  any later version.
 #
 # Entries in this file show the compile time defaults. Local configuration
-# should be created by either modifying this file, or by creating "drop-ins" in
-# the timesyncd.conf.d/ subdirectory. The latter is generally recommended.
-# Defaults can be restored by simply deleting this file and all drop-ins.
+# should be created by either modifying this file (or a copy of it placed in
+# /etc/ if the original file is shipped in /usr/), or by creating "drop-ins" in
+# /etc/systemd/timesyncd.conf.d/ directory. The latter is generally
+# recommended.  Defaults can be restored by simply deleting the main
```
Comment (YHNdnzj) [inline src/timesync/timesyncd.conf.in]: here too

**Comment by YHNdnzj** [comment]
Also, the paths in `standard-conf.xml` from your previous PR (c76f2fb0e59340222ce21f85c17d384c114db9de) should be suffixed.

**Comment by fbuihuu** [comment]
@YHNdnzj thanks for the feedback. PR updated.

**Comment by poettering** [comment]
I am sorry what? config files cannot get installed in /usr/? why would they?

i don#t follow at all anymore. can we please just get some agreement on all of this? 

this seems all so wrong. These files should never be shipped in /usr/. they are decoration, not more, hence have no place anywhere in systems without /etc/

**Comment by bluca** [comment]
Shipping defaults in /usr makes sense to me, even if they are all commented out and just decoration by default - we do sometimes patch them to change defaults. Better to patch config files than code to change embedded defaults, no?

**Comment by poettering** [comment]
i am not convinced we should ship tons of NOPs in /usr/. Systems without /etc/ don't need any of this, hence it shouldn't be there.

**Comment by poettering** [comment]
what's wrong with just shipping them in /etc/ for people who want a fully populated /etc/? people who don't need that shouldn't have that noise.

**Comment by fbuihuu** [comment]
> what's wrong with just shipping them in /etc/ for people who want a fully populated /etc/?

Shipping main config files in /etc/ is not helping users to switch to drop-ins exclusively. That's a fact, people are much more used to modifying main config files than creating drop-ins. And editing main config files, which already contain all the sections and the setting names/values is much easier.

So if one wants to customize some PID1 settings and only finds  `/etc/systemd/system.conf.d` directory (no main config file), we hope that users will be tempted to put a drop-in in the existing directory. Furthermore the fact that main config files are missing might suggest that they won't be read, hence another reason for users to prefer drop-ins.

**Comment by fbuihuu** [comment]
> Shipping defaults in /usr makes sense to me, even if they are all commented out and just decoration by default - we do sometimes patch them to change defaults. Better to patch config files than code to change embedded defaults, no?

In our case we wont patch them even if they're located in /usr because any (downstream) customization would be lost as soon as users create a main config file in /etc (regardless of the settings being changed by the user).

And in any cases drop-in is a much better solution for any other packages that would need to override some defaults. The only issue with is that drop-in in /usr can override main config files in /etc. That's the reason we're trying to switch to drop-ins exclusively and for that we need to teach users.

**Comment by fbuihuu** [comment]
> i am not convinced we should ship tons of NOPs in /usr/. Systems without /etc/ don't need any of this, hence it shouldn't be there.

The only reason to ship them in /usr is that the templates are a convenient way for users to retrieve the setting defaults. Hence if one look at the current config with `systemd-analyze cat-config` he will find them easily.

If for example `systemd-analyze cat-config --defaults` could show them, we wouldn't need to ship the templates at all.

File: src/core/user.conf.in [hunk 5]
```diff
@@ -6,9 +6,13 @@
 #  any later version.
 #
 # Entries in this file show the compile time defaults. Local configuration
-# should be created by either modifying this file, or by creating "drop-ins" in
-# the user.conf.d/ subdirectory. The latter is generally recommended.
-# Defaults can be restored by simply deleting this file and all drop-ins.
+# should be created by either modifying this file (or a copy of it placed in
+# /etc/ if the original file is shipped in /usr/), or by creating "drop-ins" in
+# /etc/systemd/user.conf.d/ directory. The latter is generally recommended.
```
Comment (keszybz) [inline src/core/user.conf.in]: "the" got lost here. In other places too.

File: src/core/system.conf.in [hunk 6]
```diff
@@ -7,10 +7,10 @@
 #
 # Entries in this file show the compile time defaults. Local configuration
 # should be created by either modifying this file (or a copy of it placed in
-# /etc if the original file is shipped in /usr), or by creating "drop-ins" in
+# /etc/ if the original file is shipped in /usr/), or by creating "drop-ins" in
 # /etc/systemd/system.conf.d/ directory. The latter is generally recommended.
 # Defaults can be restored by simply deleting the main configuration file and
-# all drop-ins located in /etc.
+# all drop-ins located in /etc/.
```
Comment (keszybz) [inline src/core/system.conf.in:13]: This chunk belongs in the other commit.

**Review by keszybz — CHANGES_REQUESTED** [review]

**Comment by keszybz** [comment]
LGTM, but we should probably first finish the discussion in https://github.com/systemd/systemd/pull/29244.

**Pull request merged**